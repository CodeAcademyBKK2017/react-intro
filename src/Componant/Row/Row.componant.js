import React, { Component } from 'react';
import Cell from '../Cell/Cell.componant';

class Row extends Component {

  render() {
    return (
      <div>
        <Cell text='x' color='#2196F3'/>
        <Cell text='x' color='#2196F3'/>
        <Cell text='x' color='#2196F3'/>
      </div>
    );
  }
}

export default Row
