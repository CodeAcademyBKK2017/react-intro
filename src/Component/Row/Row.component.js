import React from 'react'; 
import Cell from '../Cell/Cell.component'
class Row extends React.Component{
    render(){
        return(
<div style={{textAlign:'center'}}>
    <Cell willData = {this.props.data[0]} cellclickHandler={this.props.cellclickHandler}/>
    <Cell willData = {this.props.data[1]} cellclickHandler={this.props.cellclickHandler}/>
    <Cell willData = {this.props.data[2]} cellclickHandler={this.props.cellclickHandler}/>
</div>
        ) 
    }
}
export default Row;