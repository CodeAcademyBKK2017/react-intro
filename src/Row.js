import React, { Component } from 'react';
import './App.css';
import Button from './button';

class button extends Component {

  render() {
    return (
      <div>
        <Button text='hello' color='#2196F3'/>
        <Button text='hello' color='#2196F3'/>
        <Button text='hello' color='#2196F3'/>
      </div>
    );
  }
}

export default button
