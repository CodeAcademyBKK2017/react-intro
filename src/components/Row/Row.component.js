import React from 'react';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import Cell from '../Cell/Cell.component';

// class Row extends React.Component {
//   render () {
//     const {rowData, cellClickHandler} = this.props;
//     const cellLists = rowData.map((cellData, index) => <Cell key={cellData[0]} cellData={cellData[1]} color="blue" cellClickHandler={cellClickHandler} />);
//     return (
//       <div>
//         {cellLists}
//       </div>
//     );
//   }
// }

const Row = ({rowData, cellClickHandler}) => {
  const cellLists = rowData.map((cellData, index) => <Cell key={cellData[0]} cellData={cellData[1]} color="blue" cellClickHandler={cellClickHandler} />);
  return (
    <div>
      {cellLists}
    </div>
  );
};

Row.propTypes = {
  rowData: PropTypes.array.isRequired,
  cellClickHandler: PropTypes.func.isRequired
};

Row.defaultProps = {
  rowData: [],
  cellClickHandler: noop
};

export default Row;
