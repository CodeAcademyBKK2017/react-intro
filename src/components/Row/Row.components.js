import React, { Component } from 'react';
import Cell from '../Cell/Cell.components';

import PropTypes from 'prop-types';
import noop from 'lodash/noop';

class Row extends Component {
  render() {
      const style = {
        marginTop: "44px",
      };
    return (
        <div style={style}>
            <Cell name={ this.props.cellData[0] } cellClickHandler={this.props.cellClickHandler}/> <Cell name={ this.props.cellData[1] } cellClickHandler={this.props.cellClickHandler}/> <Cell name={ this.props.cellData[2] } cellClickHandler={this.props.cellClickHandler}/>
        </div>
    );
  }
}

Row.propTypes = {
  cellClickHandler: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
};
Row.defaultProps = {
  name: '',
  cellClickHandler: noop
};

export default Row;
