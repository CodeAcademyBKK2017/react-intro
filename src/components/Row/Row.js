import React, { Component } from 'react';
import Cell from '../Cell/Cell';

class Row extends Component {
  render() {
      const style = {
        marginTop: "44px",
      };
    return (
        <div style={style}>
            <Cell name="C1"/> <Cell name="C2"/> <Cell name="C3"/>
        </div>
    );
  }
}

export default Row;
