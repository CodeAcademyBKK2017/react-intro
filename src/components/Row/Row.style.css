.cell-container {

}

.cell-container + .cell-container {
    margin-top: 5px;
}