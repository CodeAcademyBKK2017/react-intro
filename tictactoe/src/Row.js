import React from 'react';

import Cell from './Cell'

class Row extends React.Component {
    render() {
        const emptyCellLists = new Array(Number(this.props.cellNumber)).fill(1);
        const cellLists = emptyCellLists.map((emptyCellList) => <Cell text="o" color="blue" />);
        return (
            <div>
                {cellLists}
            </div>
        );
    }
}

export default Row;
