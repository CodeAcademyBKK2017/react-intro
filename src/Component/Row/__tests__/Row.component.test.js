import React from 'react';
import Row from '../Row.component';
import renderer from 'react-test-renderer';

test('Row',()=>{
  const  data = ['o','x','o'];

    const tree = renderer
    .create(<Row data = "data"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});