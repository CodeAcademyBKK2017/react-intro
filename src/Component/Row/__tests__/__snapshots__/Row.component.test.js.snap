// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    onClick={undefined}
    style={
      Object {
        "background": "rgb(6, 194, 243)",
        "border": "1px solid #fff",
        "color": "#fff",
        "display": "inline-block",
        "fontSize": "20px",
        "height": "20px",
        "marginLeft": "2px",
        "padding": "25px",
      }
    }
  >
    d
     
  </div>
  <div
    onClick={undefined}
    style={
      Object {
        "background": "rgb(6, 194, 243)",
        "border": "1px solid #fff",
        "color": "#fff",
        "display": "inline-block",
        "fontSize": "20px",
        "height": "20px",
        "marginLeft": "2px",
        "padding": "25px",
      }
    }
  >
    a
     
  </div>
  <div
    onClick={undefined}
    style={
      Object {
        "background": "rgb(6, 194, 243)",
        "border": "1px solid #fff",
        "color": "#fff",
        "display": "inline-block",
        "fontSize": "20px",
        "height": "20px",
        "marginLeft": "2px",
        "padding": "25px",
      }
    }
  >
    t
     
  </div>
</div>
`;
